src/routes/language/components/ChangeLanguageModal.js

```diff
@@ -19,25 +20,35 @@ const Wrapper = styled.View`
 
 type PropsType = {
   theme: ThemeType,
-  city: string,
-  currentLanguage: string,
-  languages: Array<LanguageModel>,
+  city: string | null,
+  currentLanguage: string | null,
+  languages: Array<LanguageModel> | null,
   availableLanguages: Array<string>,
-  changeLanguage: (city: string, language: string) => void,
+  changeLanguage: (city: string, language: string) => SwitchContentLanguageActionType,
   closeModal: () => void
 }
 
 class ChangeLanguageModal extends React.Component<PropsType> {
   onPress = (model: LanguageModel) => {
-    const { closeModal, changeLanguage, city } = this.props
+    const {closeModal, changeLanguage, city} = this.props
+
+    if (!city) {
+      throw new Error('Value is unexpectedly null') // fixme: This should be handled properly if this is even possible
+    }
+
     closeModal()
     InteractionManager.runAfterInteractions(() => {
       changeLanguage(city, model.code)
     })
   }
 
   render () {
-    const { theme, languages, availableLanguages, currentLanguage } = this.props
+    const {theme, languages, availableLanguages, city, currentLanguage} = this.props
+
+    if (!languages || !currentLanguage || !city) {
```

Comment:
Or at least handle this stuff in the containers.

Reply:
We can create a task for this, as this only came up because of fixing the flow types.

But unfortunately that task will not have priority. Probably a hoc would make sense, but only if and only if the flow types work good.

The benefit of doing it in the render method is that it is very simple.

Reply:
https://issues.integreat-app.de/browse/NATIVE-53

Reply:
Old but gold task :D